gcloud/datastore/client.py

```diff
@@ -51,53 +130,168 @@ def __init__(self, dataset_id=None, namespace=None, connection=None):
         self.namespace = namespace
 
     def get(self, key, missing=None, deferred=None):
-        """Proxy to :func:`gcloud.datastore.api.get`.
+        """Retrieve an entity from a single key (if it exists).
 
-        Passes our ``dataset_id``.
+        .. note::
+
+           This is just a thin wrapper over :method:`get_multi`.
+           The backend API does not make a distinction between a single key or
+           multiple keys in a lookup request.
         """
-        return get(key, missing=missing, deferred=deferred,
-                   connection=self.connection, dataset_id=self.dataset_id)
+        entities = self.get_multi(keys=[key], missing=missing,
+                                  deferred=deferred)
+        if entities:
+            return entities[0]
 
     def get_multi(self, keys, missing=None, deferred=None):
-        """Proxy to :func:`gcloud.datastore.api.get_multi`.
+        """Retrieve entities, along with their attributes.
 
-        Passes our ``dataset_id``.
+        :type keys: list of :class:`gcloud.datastore.key.Key`
+        :param keys: The keys to be retrieved from the datastore.
+
+        :type missing: an empty list or None.
+        :param missing: If a list is passed, the key-only entities returned
+                        by the backend as "missing" will be copied into it.
+                        Use only as a keyword param.
+
+        :type deferred: an empty list or None.
+        :param deferred: If a list is passed, the keys returned
+                        by the backend as "deferred" will be copied into it.
+                        Use only as a keyword param.
         """
-        return get_multi(keys, missing=missing, deferred=deferred,
-                         connection=self.connection,
-                         dataset_id=self.dataset_id)
+        if not keys:
+            return []
+
+        ids = list(set([key.dataset_id for key in keys]))
+        if ids != [self.dataset_id]:
+            raise ValueError('Keys do not match dataset ID')
+
+        transaction = Transaction.current()
+
+        entity_pbs = _extended_lookup(
+            connection=self.connection,
+            dataset_id=self.dataset_id,
+            key_pbs=[k.to_protobuf() for k in keys],
+            missing=missing,
+            deferred=deferred,
+            transaction_id=transaction and transaction.id,
+        )
+
+        if missing is not None:
+            missing[:] = [
+                helpers.entity_from_protobuf(missed_pb)
+                for missed_pb in missing]
+
+        if deferred is not None:
+            deferred[:] = [
+                helpers.key_from_protobuf(deferred_pb)
+                for deferred_pb in deferred]
+
+        return [helpers.entity_from_protobuf(entity_pb)
+                for entity_pb in entity_pbs]
```

Comment:
Strange that the thing we used to have do this wasn't a list comprehension. Nice catch.